Modules/Core/Common/include/itkImageRegion.h

```diff
@@ -285,23 +285,18 @@ class ITK_TEMPLATE_EXPORT ImageRegion final : public Region
 
   /** Test if a region (the argument) is completely inside of this region. */
   bool
-  IsInside(const Self & region) const
+  IsInside(const Self & otherRegion) const
   {
-    IndexType beginCorner = region.GetIndex();
+    const auto otherIndex = otherRegion.m_Index;
```

Comment:
If this were "const auto & otherIndex  = otherRegion.m_index" then it would not need to make a copy of the information.  

Even better:  Remove these constant aliases, and just use "otherRegion.m_index[i]" below.

Reply:
I think the compiler is likely to optimize it away either way.

Reply:
> I think the compiler is likely to optimize it away either way.

Possible. However, I do believe the commit as a whole could significantly improve the performance, as it does less comparisons than the original code. And it appears that `ImageRegion::IsInside` is called quite frequently.